tree/dataframe/src/RLoopManager.cxx

```diff
@@ -359,6 +359,68 @@ RLoopManager::RLoopManager(std::unique_ptr<RDataSource> ds, const ColumnNames_t
    fDataSource->SetNSlots(fNSlots);
 }
 
+RLoopManager::RLoopManager(const ROOT::RDF::RDatasetSpec &&spec)
+   : fStartEntry(spec.fStartEntry), fEndEntry(spec.fEndEntry), fNSlots(RDFInternal::GetNSlots()),
+     fLoopType(ROOT::IsImplicitMTEnabled() ? ELoopType::kROOTFilesMT : ELoopType::kROOTFiles),
+     fNewSampleNotifier(fNSlots), fSampleInfos(fNSlots)
+{
+   if (spec.fTreeNames.size() == 1) {
+      // A TChain has a global name, that is the name of single tree
+      auto chain = std::make_shared<TChain>(spec.fTreeNames[0].c_str());
+      // The global name of the chain is also the name of each tree in the list
+      // of files that make the chain.
+      for (const auto &f : spec.fFileNameGlobs)
+         chain->Add(f.c_str());
+      SetTree(chain);
+   } else {
+      // Some other times, each different file has its own tree name, we need to
+      // reconstruct the full path to the tree in each file and pass that to
+      // TChain::Add
+      auto chain = std::make_shared<TChain>();
+      for (auto i = 0u; i < spec.fFileNameGlobs.size(); i++) {
+         const auto fullpath = spec.fFileNameGlobs[i] + "?#" + spec.fTreeNames[i];
+         chain->Add(fullpath.c_str());
+      }
+      SetTree(chain);
```

Comment:
the advantage of the first case is that the chain will have the tree name. In case no alias is provided, that can sometimes be helpful.

In fact, this is the same code as in https://github.com/root-project/root/blob/dea97f02bdfaf788f2c96ae2e02d20a64744364d/tree/treeplayer/src/TTreeProcessorMT.cxx#L318-L345.

Afaik, @vepadulano is working on understanding the lifetime of the friends, so that we move the quoted logic to a separate function - e.g. transform friend list info to actual friends.